src/omni_epd/displays/inky_display.py

```diff
@@ -108,19 +120,23 @@ def __init__(self, deviceName, config):
         self.width = self._device.width
         self.height = self._device.height
 
+        # get colors from the inky lib (won't be used normally as inky does conversion)
+        self.palette_filter = deviceObj.DESATURATED_PALETTE
+
     @staticmethod
     def get_supported_devices():
         result = []
 
         # python libs for this might not be installed - that's ok, return nothing
         if(InkyImpressionDisplay.check_module_installed('inky')):
             # if passed return list of devices
-            result = [f"{InkyDisplay.pkg_name}.impression"]
+            result = [f"{InkyImpressionDisplay.pkg_name}.impression"]
 
         return result
 
     def _display(self, image):
-        self._device.set_image(image)
+        # no palette adjustments need to be done as the Inky lib does them from the image
+        self._device.set_image(image.convert("RGB"), saturation=self._getfloat_device_option('saturation', .5))  # .5 is default from Inky lib

```

Comment:
Skipping the filter means `bw` mode doesn't work.

Reply:
Good catch - bw needs to be filtered first.

Reply:
Fixed. Makes sense as we're passing a full color image to the display, how would it know to do bw only? Applied the same fix for both device types, filter to BW then pass in the image.